web/src/user_profile.ts

```diff
@@ -1215,6 +1221,43 @@ export function initialize(): void {
         handle_remove_stream_subscription(target_user_id, sub, removal_success, removal_failure);
     });
 
+    $("body").on("click", "#user-profile-modal .remove-member-button", (e) => {
+        e.preventDefault();
+        const $group_row = $(e.currentTarget).closest("[data-group-id]");
+        const group_id = Number.parseInt($group_row.attr("data-group-id")!, 10);
+        const target_user_id = Number.parseInt($("#user-profile-modal").attr("data-user-id")!, 10);
+        const target_user_group = user_groups.get_user_group_from_id(group_id);
+        const $alert_box = $("#user-profile-groups-tab .group_list_info");
+
+        function removal_success(): void {
+            ui_report.success($t_html({defaultMessage: "Removed successfully!"}), $alert_box, 1200);
+        }
+
+        function removal_failure(): void {
+            let error_message;
+            if (people.is_my_user_id(target_user_id)) {
+                error_message = $t(
+                    {defaultMessage: "Error in leaving from {group_name}"},
+                    {group_name: target_user_group.name},
+                );
+            } else {
+                error_message = $t(
+                    {defaultMessage: "Error removing user from {group_name}"},
+                    {group_name: target_user_group.name},
+                );
+            }
```

Comment:
I tweaked these strings, but I wonder if we even want this function; I imagine it'll often be better to just display the error returned from the server (like "Insufficient permission" or whatever)?